### DISCUSSION TITLE
Can We migrate gatsby v2 to v5 directly

### DISCUSSION THREAD

**Discussion by chethan-b-hpe** [gatsbyjs/gatsby#38806] [Help]
We are using gatsby v2, want to migrate to v5. Is directly from v2 to v5 is possible?We are facing issues a lot.

**Comment by 808vita** [comment]
... It can get tricky and yes quite a bit of debugging.
Node v18 , webpack v5 , and many other gatsby core dependencies got breaking changes.

Good starting point is to analyse the package json of current v2 project and make a spreadsheet.
Same for the v5 gatsby as well ;  headapi , scriptapi and lots more have introduced till now.

With these two sheets check compatibility , mark necessary remarks for components and proceed slowly - fix one problematic component at a time.

I think you would benefit if you create a separate mvp using v5 gatsby and try to replicate a sample for each component and then copying over src folder & other files for testing; instead of directly trying to upgrade the whole v2 gatsby code base.


Attaching link to [gatsby release notes](https://www.gatsbyjs.com/docs/reference/release-notes/) 
Hope this helps.
Best of luck.
